### PR TITLE
Add support for synapse workspace and synapse big data pool

### PR DESCRIPTION
**Pull request by emilychu9318**
**What this PR does / why we need it**:
This PR enables the Synapse workspace and workspace bigDataPools resource in code generation and adds a test that exercises creating, updating and deleting it.

**How does this PR make you feel**:

**If applicable**:
- [:white_check_mark:] this PR contains tests

### CODE REVIEW

File: v2/internal/controllers/crd_synapse_test.go [hunk 1]
```diff
@@ -0,0 +1,117 @@
...
+	sa := &storage.StorageAccount{
+		ObjectMeta: tc.MakeObjectMetaWithName(tc.NoSpaceNamer.GenerateName("adlstore")),
+		Spec: storage.StorageAccount_Spec{
+			Location:     to.Ptr("eastus2"),
+			Owner:        testcommon.AsOwner(rg),
+			Kind:         to.Ptr(storage.StorageAccount_Kind_Spec_StorageV2),
+			Sku:          &storage.Sku{Name: to.Ptr(storage.SkuName_Standard_LRS)},
+			IsHnsEnabled: to.Ptr(true),
+			OperatorSpec: &storage.StorageAccountOperatorSpec{
+				ConfigMaps: &storage.StorageAccountOperatorConfigMaps{
```
Comment (emilychu9318) [inline v2/internal/controllers/crd_synapse_test.go:37]: ConfigMaps is implemented in: https://github.com/Azure/azure-service-operator/pull/2855

File: v2/azure-arm.yaml [hunk 2]
```diff
@@ -1601,6 +1601,28 @@ objectModelConfiguration:
...
+        AccountUrl:
+          $importConfigMapMode: optional
+      PurviewConfiguration:
+        PurviewResourceId:
+          $armReference: true
+      Workspace:
+        $export: true
+        $supportedFrom: v2.0.0
+        $azureGeneratedSecrets:
+          - SqlAdministratorLoginPassword
```
[the remark this answers is not in the document]
Comment (matthchr) [reply v2/azure-arm.yaml:1621]: This actually isn't the way to do this. To mark that property as a secret you need to do:
```
      Workspace:
        $export: true
        $supportedFrom: v2.0.0
      WorkspaceProperties:
        SqlAdministratorLoginPassword:
          $isSecret: true
```

I've sent you a PR to do this and the samples recording

File: v2/samples/synapse/v1beta/v1beta20210601_synapsebigdatapool.yaml [hunk 3]
```diff
@@ -0,0 +1,17 @@
+apiVersion: synapse.azure.com/v1beta20210601
+kind: WorkspacesBigDataPool
+metadata:
+  name: samplepool
+  namespace: default
+spec:
+  location: westcentralus
+  owner:
```
Comment (matthchr) [inline v2/samples/synapse/v1beta/v1beta20210601_synapsebigdatapool.yaml:8]: These samples need to have a recording made for them. You can do this by running the "Test_Samples_CreationAndDeletion" test. I've sent you a PR though with this already done so you don't have to wrestle through it this time.

File: v2/internal/controllers/crd_synapse_test.go [hunk 4]
```diff
@@ -0,0 +1,116 @@
...
+					Key:  accountUrlKey,
+				},
+				Filesystem: to.Ptr("default"),
+			},
+			Location: to.Ptr("eastus2"),
+			Owner:    testcommon.AsOwner(rg),
+			Tags:     map[string]string{"cheese": "blue"},
+		},
+	}
+
```
Comment (matthchr) [inline v2/internal/controllers/crd_synapse_test.go:65]: minor: Our test style is usually to create the resources associated with the test here, and then only create the sub-resources in the sub-test. That way the sub-test is focused on the resources for its name, and the parent test is focused on the resources related to its name.

I've sent you a PR with this minor refactor.

**Review by matthchr — COMMENTED** [review]
This looks almost there to me. I had a few really small comments but figured rather than make you do them, since we're about to release and it'd be great to get this change in before we do, I sent you a PR with the changes I requested: https://github.com/emilychu9318/azure-service-operator/pull/1

If you're happy with those changes, merge them in and we should be able to get this merged. I'll hold off approval until you do but once that stuff is merged this LGTM.

**Comment by emilychu9318** [comment]
Thank you for making the PR to fix these issues!

**Comment by emilychu9318** [comment]
> @microsoft-github-policy-service agree

@microsoft-github-policy-service agree

**Pull request merged**